include/dca/math/nfft/nfft_helper.cu.hpp

```diff
@@ -0,0 +1,184 @@
+             sizeof(ScalarType) * parameters_real_host.size(), cudaMemcpyHostToDevice);
+}
+
+template <typename ScalarType>
+__device__ int NfftHelper<ScalarType>::deltaR(const int r1, const int r2) const {
+  const int ld = parameters_int_[2];
+  return sub_matrix_[r1 + ld * r2];
+}
+
+template <typename ScalarType>
+__host__ void NfftHelperManager<ScalarType>::set(const int nb, const int nr, const int* sub_r,
+                                                 int lds, int oversampling, int window_sampling,
+                                                 ScalarType t0, ScalarType delta_t,
+                                                 const ScalarType t0_window,
+                                                 const ScalarType delta_t_window,
+                                                 const ScalarType beta) {
+  NfftHelper<ScalarType>::set(nb, nr, sub_r, lds, oversampling, window_sampling, t0, delta_t,
+                              t0_window, delta_t_window, beta);
+  oversampling_ = oversampling;
+}
+
+template <typename ScalarType>
+__device__ int inline NfftHelper<ScalarType>::computeLinearIndex(const int b1, const int b2,
+                                                                 const int r1, const int r2) const {
+  const int delta_r = deltaR(r2, r1);
+  const int nb = parameters_int_[0];
+  return b1 + b2 * nb + delta_r * nb * nb;
+}
+
+template <typename ScalarType>
+__device__ ScalarType NfftHelper<ScalarType>::computeTau(ScalarType t1, ScalarType t2) const {
+  const ScalarType tau_scale = parameters_real_[6];
+  return (t1 - t2) * tau_scale;
+}
+
+template <typename ScalarType>
+__device__ void NfftHelper<ScalarType>::computeInterpolationIndices(ScalarType t, int& t_idx,
+                                                                    int& conv_coeff_idx,
+                                                                    ScalarType& delta_t) const {
+  const ScalarType t0 = parameters_real_[0];
+  const ScalarType delta_t_padded = parameters_real_[1];
+  const ScalarType t0_window = parameters_real_[2];
+  const ScalarType delta_t_window = parameters_real_[3];
+  auto get_tau = [=](int index) { return t0 + index * delta_t_padded; };
+  auto get_window_tau = [&](int index) { return t0_window + index * delta_t_window; };
+  const ScalarType one_div_delta = parameters_real_[4];
+  const ScalarType one_div_delta_wnd = parameters_real_[5];
+
+  t_idx = (t - t0) * one_div_delta;
+  const int t_wnd_idx = (get_tau(t_idx) - t - t0_window) * one_div_delta_wnd;
+  delta_t = get_tau(t_idx) - t - get_window_tau(t_wnd_idx);
+
+  const int oversampling = parameters_int_[3];
+  const int window_sampling = parameters_int_[4];
+  t_idx -= oversampling;
+  const int delta_tau_index = t_wnd_idx - oversampling * window_sampling;
+  const int j = delta_tau_index % window_sampling;
+  const int i = (delta_tau_index - j) / window_sampling;
+
+  constexpr int interpolation_size = 4;  // Hard coded cubic case.
```

Comment:
This hard coding might be prone to error in the future.  My suggestion is better to make it dependent on the NfftModeNames.

Reply:
used a template parameter in aefb57e04087480b2a95aad4a28b1f22cfb3a435 nad made clear that the kernels are implemented only for the cubic case. Note that the DnfftGpu is defined only for the CUBIC case.